catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/HistoryMetacardImpl.java

```diff
@@ -0,0 +1,248 @@
+    public static final String STATE = "state";
+
+    /**
+     * {@link ddf.catalog.data.Attribute} name for the editor of this {@link Metacard} revision.
+     *
+     * @since DDF-2.9.0
+     */
+    public static final String EDITED_BY = "edited-by";
+
+    /**
+     * {@link ddf.catalog.data.Attribute} name for version date of this {@link Metacard} revision.
+     *
+     * @since DDF-2.9.0
+     */
+    public static final String VERSIONED = "versioned";
+
+    /**
+     * {@link ddf.catalog.data.Attribute} name for metacard ID on a history item of this {@link Metacard} revision.
+     *
+     * @since DDF-2.9.0
+     */
+    public static final String ID_HISTORY = "history-id";
+
+    public static final String TAGS_HISTORY = "history-tags";
+
+    private static MetacardType versionHistoryMetacard;
+
+    public static final String METACARD_TYPE_NAME = "ddf.history.metacard";
+
+    static {
+        HashSet<AttributeDescriptor> historyDescriptors =
+                new HashSet<>(BasicTypes.BASIC_METACARD.getAttributeDescriptors());
+        historyDescriptors.add(new AttributeDescriptorImpl(STATE,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(EDITED_BY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(VERSIONED,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.DATE_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(ID_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                false /* multivalued */,
+                BasicTypes.STRING_TYPE));
+        historyDescriptors.add(new AttributeDescriptorImpl(TAGS_HISTORY,
+                true /* indexed */,
+                true /* stored */,
+                false /* tokenized */,
+                true /* multivalued */,
+                BasicTypes.STRING_TYPE));
+
+        versionHistoryMetacard = new MetacardTypeImpl(METACARD_TYPE_NAME, historyDescriptors);
+    }
+
+    /**
+     * Will convert the given {@link Metacard} to a {@link HistoryMetacardImpl} by cloning
+     * it and adding the current subject, time, and a random UUID. Cannot take a
+     * {@link HistoryMetacardImpl} as the sourceMetacard.
+     *
+     * @param sourceMetacard Metacard to clone and create a history item from
+     * @param action         Which action was done to modify the metacard
+     */
+    public HistoryMetacardImpl(Metacard sourceMetacard, Action action)
```

Comment:
I'd argue that unchecked exceptions should be declared in the javadoc and left out of the method signature.